src/js/index.js

```diff
@@ -0,0 +1,331 @@
+    for (var val2 = val+1; val2 < len; val2++){
+      if (inputImageArray[val2][toSortBy] <= inputImageArray[val][toSortBy]){
+        var temp = inputImageArray[val2];
+        inputImageArray[val2] = inputImageArray[val];
+        inputImageArray[val] = temp;
+      }
+    }
+  }
+  console.log('Returning image array:', inputImageArray);
+  console.log('Sorted the Gallery;');
+  return inputImageArray;
+}
+
+function filterGallery(toFilterBy, inputImageArray){
+  currentFilterValue = toFilterBy;
+  filterByActive = true;
+
+  var outputImages = [];
+  var len = inputImageArray.length;
+
+  for (var val=0; val < len; val++){
+    if(inputImageArray[val]['category']===toFilterBy){
+      outputImages.push(inputImageArray[val]);
+    }
+  }
+
+  return outputImages;
+}
+
+function changeModalImage(type){
+
+  console.log('Change modal image:');
+  console.log(currentImagesData);
+  console.log('Change modal image;');
+
+  var currentModalLength = currentImagesData.length;
+
+  var modalImage = document.getElementById('modal-image');
+  var newSrc = '';
+  
+  var newIndex = parseInt(currentModalIndex)+ parseInt(type);
+  if ( type == 1){
+    console.log('Advance by 1 :)');
+    if (newIndex < currentModalLength){
+      newSrc = newIndex;
+    } else {
+      newSrc = 0;
+    }
+  } else{
+    console.log('Advance by -1 :)');
+    if( newIndex > 0){
+      newSrc = newIndex;
+    } else {
+      newSrc = currentModalLength;
+    }
+  }
+  currentModalIndex = newSrc;
+
+  var newImageSrc = currentImagesData[newSrc].src;
+  modalImage.setAttribute('src',newImageSrc)
+}
+
+function closeModal (){
+  var modalContainer = document.getElementById('gallery-modal');
+  modalContainer.classList.add('hide');
+}
+
+function showImageModal(e){
+  console.log(e);
+  console.log(e.target);
+  console.log(e.target.src);
+
+  var modalContainer = document.getElementById('gallery-modal');
+  modalContainer.classList.remove('hide');
+  var modalImage = document.getElementById('modal-image');
+  modalImage.setAttribute('src', e.target.src);
+  currentModalIndex = e.target.dataset.position;
+}
+
+function buildModalUI(){
+  var modalContainer = document.createElement('div');
+  modalContainer.setAttribute('class', 'modal-container hide');
+  modalContainer.setAttribute('id','gallery-modal');
+
+  var modalOverlay = document.createElement('div');
+  modalOverlay.setAttribute('class','modal-overlay');
+
+  var modal = document.createElement('div');
+  modal.setAttribute('class', 'modal');
+
+  var modalImage = document.createElement('img');
+  modalImage.setAttribute('class', 'modal-image');
+  modalImage.setAttribute('id', 'modal-image');
+  modalImage.setAttribute('src', 'images/1.jpg');
+
+  var crossButton = document.createElement('div');
+  crossButton.setAttribute('id', 'close-modal');
+  crossButton.setAttribute('class', 'close-modal');
+  crossButton.innerHTML = '&#10005;';
+  crossButton.addEventListener("click", closeModal)
```

Comment:
You can read on what's the 3rd argument of addEventListener and what's the use case, will be fun